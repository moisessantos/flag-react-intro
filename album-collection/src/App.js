import React, { Component } from 'react';
import Album from './Album';
import styles from './App.module.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      data: [],
      users: []
    }
  }

  componentDidMount() {
    fetch("https://jsonplaceholder.typicode.com/albums")
      .then((response) => response.json())
      .then((data) => {
        this.setState({
          data
        })
      });
  }

  getUserById(id) {
    const newUsers = this.state.users;
    fetch("https://jsonplaceholder.typicode.com/users/" + id)
      .then((response) => response.json())
      .then((data) => {
        newUsers.push(data)
        this.setState({
          users: newUsers
        })
      });
  }

  render() {
    return (
      <div className={styles.container}>
        <ul>
          {this.state.data.map((album) => (
            <Album
              key={album.id}
              album={album}
              fetchUser={() => this.getUserById(album.userId)}
              authorName={this.state.users.find(author => author.id === album.userId)?.name}
            />)
          )}
        </ul>
      </div>
    );
  }
}

export default App;
