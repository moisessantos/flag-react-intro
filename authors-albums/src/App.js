import React from "react";
import UserList from "./users-list";
import AlbumList from "./albums-list";

class App extends React {
  constructor(){
    super();
    this.state = {
      users: [],
      albumsByUserId: {},
      currentUserId: undefined,
      //albums: []
    }
    this.getAlbumsByUserId = this.getAlbumsByUserId.bind(this);
  }

  /*albumsByUserId: {
    1: [ lista de albums],
    2: [ lista de albums]
  }*/

  componentDidMount(){
    fetch("https://jsonplaceholder.typicode.com/users")
      .then(response => response.json())
      .then(users => {
        this.setState({ users });
      })
  }

  getAlbumsByUserId(userId) {
    if(this.state.albumsByUserId[userId]){
      this.setState({
        currentUserId: userId
      })
      return;
    }
    fetch("https://jsonplaceholder.typicode.com/albums?userId=" + userId)
      .then(response => response.json())
      .then(albums => {
        const albumsByUserId = this.state.albumsByUserId;
        albumsByUserId[userId] = albums;
        this.setState({ albumsByUserId, currentUserId: userId });
      })
  }
  
  render() {
    return (
      <div className="App">
        <UserList users={this.state.users} handleClick={this.getAlbumsByUserId} />
        <br />
        {
          this.state.currentUserId && 
            <AlbumList albums={this.state.albumsByUserId[this.state.currentUserId]} />}
      </div>
    );
  }
}

export default App;
