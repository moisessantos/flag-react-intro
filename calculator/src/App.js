import React, { Component } from 'react';

import Result from './components/Result';
import { KeyPad } from './components';
import { ClickContext } from './ClickContext';

class App extends Component {
  constructor(){
      super();

      this.state = {
          result: ""
      }
  }

  onClick = button => {
      if(button === "="){
          this.calculate();
      } else if(button === "C"){
            this.reset();
      } else if(button === "CE"){
          this.backspace();
      } else {
          this.setState({ result: this.state.result + button });
      }
  };

  reset = () => {
    this.setState({ result: "" });
  };

  calculate = () => {
      let { result } = this.state;
      result = this.state.result.includes('--') ? result.replace('--','+') : result;

      try {
          this.setState({
              // eslint-disable-next-line
              result: eval(result) + ""
          })
      } catch (e) {
          this.setState({
              result: "error"
          })
      }
  };

  backspace = () => {
      this.setState({
          result: this.state.result.slice(0, -1)
      })
  };

  render() {
      return (
          <div>
              <div className="calculator-body">
                  <h1>Simple Calculator</h1>
                  <Result>{this.state.result}</Result>
                  <ClickContext.Provider value={this.onClick}>
                    <KeyPad />
                  </ClickContext.Provider>
              </div>
          </div>
      );
  }
}

export default App;