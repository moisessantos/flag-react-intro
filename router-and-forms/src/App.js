import { BrowserRouter, Routes, Route, Link } from "react-router-dom";
import { Home, NotFound, ReactForm, FormikForm, HookForm } from "./pages";
import routes from "./routes";

function App() {
  return (
      <BrowserRouter>
        <nav>
          <Link to={routes.Home}>Home</Link>
        </nav>
        <Routes>
          <Route exact path={routes.ReactForm} element={<ReactForm />}>
            <Route path=":name" element={<ReactForm />} />
          </Route>
          <Route path={routes.FormikForm} element={<FormikForm />} />
          <Route path={routes.HookForm} element={<HookForm />}>
            <Route path=":name" element={<HookForm />} />
          </Route>
          <Route index element={<Home />} />
          <Route path="*" element={<NotFound />} />
        </Routes>
      </BrowserRouter>
  );
}

export default App;
