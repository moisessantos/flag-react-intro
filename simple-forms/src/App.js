import React from "react";
import {
  BrowserRouter,
  Routes,
  Route
} from "react-router-dom";

import ReactForm from "./React-form";
import FormikForm from "./Formik-form";
import Home from './Home';
import NotFound from './NotFound';

const App = () => (
  <BrowserRouter>
    <Routes>
      <Route path="/react-form" element={<ReactForm />} />
      <Route path="/formik-form" element={<FormikForm />} />
      <Route index element={<Home />} />
      <Route path="*" element={<NotFound />} />
    </Routes>
  </BrowserRouter>
);

export default App;
