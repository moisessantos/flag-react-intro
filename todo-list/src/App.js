import React from 'react';
import styles from './App.module.css';

class App extends React.Component {
    constructor() {
        super();
        this.state = {
            todos: ["Vestir roupa", "Lavar os dentes", "Ir Ã s compras", "Ir ao shopping"],
            value: ''
        };
        this.addTodo = this.addTodo.bind(this);
        this.changeValue = this.changeValue.bind(this);
        this.deleteTodo = this.deleteTodo.bind(this);
    }

    changeValue(e){
        this.setState({
            todos: this.state.todos,
            value: this.state.value + e.target.value
        })
    }

    deleteTodo(todo) {
        const newTodos = this.state.todos;
        newTodos.pop();
        this.setState({
            todos: newTodos,
        });
    }

    addTodo() {
        const newTodos = this.state.todos;
        newTodos.push(this.state.value);
        this.setState({
            todos: newTodos,
            value: ''
        });
    }

    render() {
        return (
            <div className={styles.container}>
                <h1>A minha lista de afazeres</h1>
                <ul>
                    {this.state.todos.map((valor) => {
                        return (
                            <li key={valor}>
                                <span>{valor}</span>
                                <button onClick={() => this.deleteTodo("")}>Delete</button>
                            </li>
                        );
                    })}
                </ul>
                <input type="text" value={this.state.value} onChange={this.changeValue} />
                <button onClick={this.addTodo}>Add</button>
            </div >
        );
    }
}

export default App;