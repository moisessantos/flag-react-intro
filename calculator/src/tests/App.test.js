import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

describe('App tests', () => {
    it('should render as expected', () => {
        const wrapper = shallow(<App />);
        expect(wrapper).toMatchSnapshot();
    });

    it.each(
        [
            ["calculate", "="],
            ["reset","C"],
            ["backspace", "CE"]
        ],
        'onClick should call %s when %s is pressed'
        ,(func, val) => {
            const wrapper = shallow(<App />);
            const spy = jest.spy(wrapper.instance(), func);

            wrapper.instance().onClick(val);

            expect(spy).toBeCalledTimes(1);
        });
    
    it('onClick should append to state when a number or operator is pressed', () => {
        const wrapper = shallow(<App />);
        wrapper.setState({ result: "1" })
        wrapper.instance().onClick("+");

        expect(wrapper.state().result).toBe("1+");
    });

    it('reset should reset the result', () => {
        const wrapper = shallow(<App />);
        wrapper.setState({ result: "1" })
        wrapper.instance().reset();

        expect(wrapper.state().result).toBe("");
    });

    
    it('backspace should remove the last char from the result', () => {
        const wrapper = shallow(<App />);
        wrapper.setState({ result: "1+" })
        wrapper.instance().backspace();

        expect(wrapper.state().result).toBe("1");
    });

    it('calculate should do the calculation in the result', () => {
        const wrapper = shallow(<App />);
        wrapper.setState({ result: "1+2" })
        wrapper.instance().calculate();

        expect(wrapper.state().result).toBe("3");
    });

    it('calculate should do the calculation when -- is present in the result', () => {
        const wrapper = shallow(<App />);
        wrapper.setState({ result: "1--2" })
        wrapper.instance().calculate();

        expect(wrapper.state().result).toBe("3");
    });

    it('calculate should return error in case of typos 2', () => {
        const wrapper = shallow(<App />);
        wrapper.find('ContextProvider').props().value(1);
        wrapper.find('ContextProvider').props().value("+");
        wrapper.find('ContextProvider').props().value("*");
        wrapper.find('ContextProvider').props().value(2);
        wrapper.find('ContextProvider').props().value("=");
        expect(wrapper.find('Result').props().children).toBe("error");
    });

    it('resets or backspaces result', () => {
        const wrapper = shallow(<App />);
        wrapper.find('ContextProvider').props().value(1);
        wrapper.find('ContextProvider').props().value("C");
        expect(wrapper.find('Result').props().children).toBe("");
        wrapper.find('ContextProvider').props().value(2);
        wrapper.find('ContextProvider').props().value("*");
        wrapper.find('ContextProvider').props().value("CE");
        expect(wrapper.find('Result').props().children).toBe("2");
    });
})