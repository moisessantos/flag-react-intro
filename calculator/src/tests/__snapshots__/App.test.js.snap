// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should render as expected 1`] = `
<div>
  <div
    className="calculator-body"
  >
    <h1>
      Simple Calculator
    </h1>
    <Result />
    <ContextProvider
      value={[Function]}
    >
      <Memo(KeyPad) />
    </ContextProvider>
  </div>
</div>
`;
